server/channels/app/channel_test.go

```diff
@@ -247,6 +247,54 @@ func TestMoveChannel(t *testing.T) {
 		require.Equal(t, model.SidebarCategoryChannels, categories.Categories[1].Type)
 		assert.Contains(t, categories.Categories[1].Channels, channel.Id)
 	})
+
+	t.Run("should update threads when moving channels between teams", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+
+		sourceTeam := th.CreateTeam()
+		targetTeam := th.CreateTeam()
+		channel := th.CreateChannel(th.Context, sourceTeam)
+
+		th.LinkUserToTeam(th.BasicUser, sourceTeam)
+		th.LinkUserToTeam(th.BasicUser, targetTeam)
+		th.AddUserToChannel(th.BasicUser, channel)
+
+		// Create a thread in the channel
+		post := &model.Post{
+			UserId:    th.BasicUser.Id,
+			ChannelId: channel.Id,
+			Message:   "test",
+		}
+		post, appErr := th.App.CreatePost(th.Context, post, channel, model.CreatePostFlags{})
+		require.Nil(t, appErr)
+
+		// Post a reply to the thread
+		reply := &model.Post{
+			UserId:    th.BasicUser.Id,
+			ChannelId: channel.Id,
+			RootId:    post.Id,
+			Message:   "reply",
+		}
+		_, appErr = th.App.CreatePost(th.Context, reply, channel, model.CreatePostFlags{})
+		require.Nil(t, appErr)
+
+		// Check that the thread count before move
+		threads, appErr := th.App.GetThreadsForUser(th.BasicUser.Id, targetTeam.Id, model.GetUserThreadsOpts{})
+		require.Nil(t, appErr)
+
+		require.Zero(t, threads.Total)
+
+		// Move the channel to the target team
+		appErr = th.App.MoveChannel(th.Context, targetTeam, channel, th.BasicUser)
+		require.Nil(t, appErr)
+
+		// Check that the thread was moved
+		threads, appErr = th.App.GetThreadsForUser(th.BasicUser.Id, targetTeam.Id, model.GetUserThreadsOpts{})
+		require.Nil(t, appErr)
+
+		require.Equal(t, int64(1), threads.Total)
```

Comment:
Just to be completionist, can we test that before the move the source had 1 thread, and after the move the source had 0 threads?